dds/DCPS/RTPS/ICE/AgentImpl.h

```diff
@@ -35,9 +35,9 @@ typedef std::vector<FoundationType> FoundationList;
 
 class AgentImpl
   : public virtual Agent
-  , public virtual DCPS::ReactorInterceptor
   , public virtual DCPS::ShutdownListener
   , public virtual DCPS::InternalDataReaderListener<DCPS::NetworkInterfaceAddress>
+  , public DCPS::ReactorInterceptor
```

Comment:
Is the order important?  @jrw972

Reply:
It was complaining about the initialization order between InternalDataReaderListener and Reactor Interceptor after I made this change. Evidently virtual inheritance gets initialized in a different order?

Reply:
Yes, it's on this page https://en.cppreference.com/w/cpp/language/constructor scroll down to `Initialization order`